pkg/querier/tail.go

```diff
@@ -52,6 +52,7 @@ type Tailer struct {
 	querierTailClientsMtx sync.RWMutex
 
 	stopped          bool
+	stoppedMtx       sync.RWMutex
```

Comment:
Rather than a mutex here, what do you think of using `atomic.Bool`? That way, there's less dealing with mutexes and anyone who needs to use `stopped` in the future knows it's synchronized